pkg/signal/debugtrap.go

```diff
@@ -0,0 +1,18 @@
+// +build !windows
+
+package signal
+
+import (
+	"os"
+	gosignal "os/signal"
+	"syscall"
+)
+
+func setupSigusr1Trap() {
+	c := make(chan os.Signal, 1)
+	gosignal.Notify(c, syscall.SIGUSR1)
+	go func() {
+		<-c
```

Comment:
I think we're missing a loop of some sort here, because the way it's written you can only dump the stack once here.